src/librustdoc/doctest/make.rs

```diff
@@ -339,9 +338,6 @@ fn parse_source(source: &str, crate_name: &Option<&str>) -> Result<ParseSourceIn
         *prev_span_hi = hi;
     }
 
-    // Recurse through functions body. It is necessary because the doctest source code is
-    // wrapped in a function to limit the number of AST errors. If we don't recurse into
-    // functions, we would thing all top-level items (so basically nothing).
```

Comment:
Outdated since #138104 when I suggested to remove the recursion in https://github.com/rust-lang/rust/pull/138104#discussion_r2015246273.